src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -498,8 +507,12 @@ private void HandleFinished(bool success, ClientSideStatus receivedStatus)
                 ReleaseResourcesIfPossible();
             }
 
-            var status = receivedStatus.Status;
+            if (delayedStreamingWriteTcs != null)
```

Comment:
as `delayedStreamingWriteTcs` is set under the lock in `HandleSendFinished`, should it be checked/filled under the lock here as well?